docs/API/metadata.md

```diff
@@ -0,0 +1,43 @@
+
+The Pbench server will also calculate a default deletion date for the dataset based on the owner's retention policy and the server administrator's retention policy along with some other internal management context. These are all accessible under the `server` metadata key namespace.
+
+Clients can also set arbitrary metadata through the "dashboard" and "user" metadata namespaces:
+* The "dashboard" namespace can only be modified by the owner of the dataset, and is visible to anyone with read access to the dataset.
+* The "user" namespace is private to each authenticated user, and even if you don't own a dataset you can set your own private "user" metadata to help you categorize that dataset and to find it again.
+
+Metadata namespaces are hierarchical, and are exposed as nested JSON objects. You can address an entire namespace, e.g., `dashboard` or `dataset` and retrieve the entire JSON object, or you can address nested objects or values using a dotted metadata key path like `dashboard.contact.email` or `dataset.metalog.pbench.script`.
+
+## Metadata namespaces
+
+There are currently three metadata namespaces.
+
+The `dataset` and `server` namespaces are defined and managed by Pbench. The `dashboard` namespace allows an authenticated client to define an arbitrary nested set of JSON objects. The `user` namespace is similar; except that where metadata in the `dashboard` namespace can only be modified by the owner of the dataset, and is visible to all clients with read access to the dataset, any authenticated user can set arbitrary values in the `user` namespace and those values are visible only to the same authenticated user.
+
+### Dataset namespace
+
+This defines the dataset resource, and contains metadata from the dataset tarball recieved from the Pbench agent, including the full contents of a `metadata.log` file which was generated when the agent packaged up the result data into a tarball.
+
+This namespace includes the resource name, which can be modified by the owner of the dataset, but all other metadata in this namespace is controlled by the server.
+
+### Server namespace
+
+This defines internal server state related to a dataset that's not inherent to the representation of the data. Most of this is not of much use to external clients, but can be observed.
+
+Only `server.deletion` can be modified by an authenticated client on behalf of the owner of the dataset. When a dataset is uploaded, the server computes an expected time representing the user's default data retention period based on the time the dataset was uploaded. The owner of the dataset is allowed to modify this later, as long as the new timestamp is within the maximum allowed server data retention period.
```

Comment:
Similar to @webbnh's observations, when I read the first couple of sentences it seems like we were heading in the direction of exposing relevant server settings as immutable metadata.  With the notion of the owner allowed to modify it, this would seem to fit better as metadata for the dataset.

Reply:
OK, I keep loosing my grip on this stuff (which is a "me problem", mostly).  So, we have "four" namespaces (at the moment, and ignoring "server configuration" and "user profile"), and they _all_ are attached to a dataset, such that the same keywords potentially have different values when applied to different datasets.  Yes?

If that's the case, then I'm way less concerned about `server.deletion` and the resource name.  I mean, I do think it would be nice if there weren't namespaces which were immutable except for one key, but that's much less of an issue than having namespaces in which some keys are tied to datasets and some keys aren't.